### DISCUSSION TITLE
How to add a sponsored bounty to a critical bug?

### DISCUSSION THREAD

**Discussion by robertswan** [pixijs/pixijs#7880] [Q&A] [answered]
Hi,

How would I proceed to discuss a sponsored bounty for a bug? Several of our products are showing bugs on ios15 (all new iPhones plus OS upgraders) that look related to clipping. 

https://github.com/pixijs/pixijs/issues/7855 looks very similar to it. 

- In our products we use pixi 4.5.3 which shows a similar bug. 
- I'm told that upgrading to 5.3.9 (perhaps all of 5?) has the bug removed. I haven't verified this yet as an exhaustive fix.
- The previously existing bug above reports the issue in 6.0.0 and up. 

Upgrading to 5.3.9 is a big risk, and the on/off nature across versions marks the bug as volatile so I'm wary of pursuing the upgrade.

We have another route - hack the viewport so it is treated as bigger. I don't have details of the teams proposed implementation, but likely this could cause other knock on issues.

Ideally the root cause is found and patch applied to 4.5.3, and then checked for applicability to 6. For any sponsored bug I would collect more details and investigations but we're hoping to move fast to enable the resumption of our most popular products.

Rob

**Comment by bigtimebuddy** [comment]
Thanks for being willing to help squash these issues. We've been plagued by this Safari iOS release.

Here's some info about sponsoring bounties: https://github.com/pixijs/pixijs/wiki/Bounties#--sponsoring

Happy to work with you on it. First things first, please make sure we have good issues filed for all issues (with reproductions). Preference is for v6 to fix first and then backport as needed.

**Comment by ivanpopelyshev** [comment]
> Upgrading to 5.3.9 is a big risk, and the on/off nature across versions marks the bug as volatile so I'm wary of pursuing the upgrade.

Try upgrade, v5 is LTS. We can support it better than v4. Spend money on it if you need help.

> the on/off nature across versions

so one of new mask patches in v6 changed the behaviour, might be something with new projectionSystem implementation.

**Comment by robertswan** [comment, marked as answer]
With the amount of software affected by this bug any upgrade rather than hotfix will take an extra week of QA at least, which is a big financial problem. We are investigating this anyway but I have to cover alternatives.

I need to offer a bug bounty on the older version otherwise there's no gain to us. If that is against the processes here I respect that and will not do that. However, if I can offer the bounty on the older version then once fix is in progress I'll commit to put a bounty on the v6 version, even though our projects won't benefit from this.

Please let me know.

**Reply by bigtimebuddy** [reply]
Understood. Go ahead and offer a bounty for the older version (I can tag the issue). As long as you agree to payout when a fix is submitted, then no harm in posting now. I'd suggest setting the initial bounty high because v4 is very old at this point and it might help get a fix. How much were you thinking?

**Reply by robertswan** [reply]
I'm shooting in the dark, so figured 1000GBP (1300USD) bounty + 10% on top for v4 bug. Please let me know if I'm off the mark.

**Reply by robertswan** [reply]
https://github.com/pixijs/pixijs/issues/7855#issuecomment-945922918
Hope this is what you intend. I'll transfer the bounties this evening + 10%. Will be from my account vs company as I can see there's work on a fix already and the money will be received faster this way.

**Reply by robertswan** [reply]
Bounties transferred in one payment and showing up on opencollective.

**Comment by ivanpopelyshev** [comment]
Can you please try `renderer.maskManager.enableScissor=false` ?

**Reply by jmlee2k** [reply]
You mean renderer.mask.enableScissor, right? :)

**Reply by ivanpopelyshev** [reply]
v4.x

**Reply by jmlee2k** [reply]
ah, my bad!

**Reply by ivanpopelyshev** [reply]
Basically, https://bugs.webkit.org/show_bug.cgi?id=229989 - that's it. When scissor width height is zero , we have this result on some videocards. So, we have to track it and dont render subtree at all in that case

**Reply by robertswan** [reply]
I've passed on renderer.maskManager.enableScissor=false and https://github.com/pixijs/pixijs/pull/7888 to the team. They are in another time zone so feedback will likely be slow. Will try to hack these things into the products myself.

**Reply by ivanpopelyshev** [reply]
I and @eXponenta are working on real fix. `enableScissor=false` is a fast workaround, if it works for you then its good, but disabling scissor  has bad aftertaste, we have to fix everything properly :)

**Reply by jmlee2k** [reply]
Of course, a proper fix is best, but it's good to know that `enableScissor=false` is a viable workaround.  Thanks for the info!
